components/core/CMakeLists.txt

```diff
@@ -0,0 +1,36 @@
+# Copyright (c) 2023, Thomas Atkinson
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 the "License";
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+vkb__register_component(
+    NAME core
+    HEADERS
+        include/core/util/strings.hpp
+        include/core/util/error.hpp
+        include/core/util/hash.hpp
```

Comment:
Why is there an additional folder level named "util"?

Reply:
What I found with the v2.0 branch was the weird multiple nested components. This was mentioned on one of the PRs but I decided to address it at a different point.

The aim of core is to contain all the interfaces used throughout the rest of the components. 

```
core/window.hpp
core/model_loader.hpp
core/image_loader.hpp
```

and so on...

This stops the multiple nested components issue seen in the assets component on the v2.0 branch

The aim is to keep this header and abstract interface only... however, its likely it might evolve

Reply:
Util has landed there for now as it is common to have core utilities used throughout all the components which dont take up much space and don't warrant their own component